app/models/container/purging.rb

```diff
@@ -0,0 +1,5 @@
+class Container < ApplicationRecord
+  module Purging
+    include ArchivedPurgingMixin
+  end
+end
```

Comment:
I don't think you need these layers of indirection.  I think you can just `include ArchivedPurgingMixin` directly into the model.